marketplace/cloudinary-assets/src/index.js

```diff
@@ -0,0 +1,104 @@
+import { Cloudinary as cloudinaryCore } from 'cloudinary-core';
+
+import { setup } from 'shared-dam-app';
+
+import logo from './logo.svg';
+import descriptor from '../extension.json';
+
+const VALID_IMAGE_FORMATS = ['svg', 'jpg', 'png', 'gif', 'jpeg'];
+const MAX_FILES_UPPER_LIMIT = 25;
+const CTA = 'Select or upload a file on Cloudinary';
+
+function makeThumbnail(resource, config) {
+  const cloudinary = new cloudinaryCore({
+    cloud_name: config.cloudName,
+    api_key: config.apiKey
+  });
+
+  let url;
+  const alt = [resource.public_id, ...(resource.tags || [])].join(', ');
+  const transformations = { width: 150, height: 100, crop: 'fill' };
+
+  if (resource.resource_type === 'image' && VALID_IMAGE_FORMATS.includes(resource.format)) {
+    url = cloudinary.url(resource.public_id, transformations);
+  } else if (resource.resource_type === 'video') {
+    url = cloudinary.video_thumbnail_url(resource.public_id, transformations);
+  }
+
+  return [url, alt];
+}
+
+function renderDialog(sdk) {
+  const { cloudinary } = window;
+  const config = sdk.parameters.invocation;
+
+  const options = {
+    cloud_name: config.cloudName,
+    api_key: config.apiKey,
+    max_files: config.maxFiles,
+    multiple: config.maxFiles > 1,
+    inline_container: '#root',
+    remove_header: true
+  };
+
+  const instance = cloudinary.createMediaLibrary(options, {
+    insertHandler: data => sdk.close(data)
+  });
+
+  instance.show();
+
+  sdk.window.updateHeight(window.outerHeight);
+}
+
+async function openDialog(sdk, currentValue, config) {
+  const maxFiles = config.maxFiles - currentValue.length;
```

Comment:
If this value is `<= 0` should we actually disable the dialog?

Reply:
This is done in `isDisabled`, if it returns `true` it won't be even possible to end up in this place.